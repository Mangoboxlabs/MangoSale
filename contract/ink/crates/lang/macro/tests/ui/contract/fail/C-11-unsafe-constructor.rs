use ink_lang as ink;

#[ink::contract]
mod noop {
    #[ink(storage)]
    pub struct Noop {}

    impl Noop {
        #[ink(constructor)]
        pub unsafe fn unsafe_constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn noop(&self) {}
    }
}

fn main() {}
