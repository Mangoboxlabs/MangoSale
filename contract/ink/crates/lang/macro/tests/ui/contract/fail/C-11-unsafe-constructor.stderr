error: ink! constructors must not be unsafe
  --> $DIR/C-11-unsafe-constructor.rs:10:13
   |
10 |         pub unsafe fn unsafe_constructor() -> Self {
   |             ^^^^^^
